source/extensions/access_loggers/grpc/http_grpc_access_log_impl.cc

```diff
@@ -135,6 +143,14 @@ void HttpGrpcAccessLog::emitLog(const Http::RequestHeaderMap& request_headers,
   }
   response_properties->set_response_headers_bytes(response_headers.byteSize());
   response_properties->set_response_body_bytes(stream_info.bytesSent());
+  if (stream_info.getDownstreamBytesMeter() != nullptr) {
+    response_properties->set_downstream_header_bytes_sent(
+        stream_info.getDownstreamBytesMeter()->headerBytesSent());
+  }
+  if (stream_info.getUpstreamBytesMeter() != nullptr) {
+    response_properties->set_upstream_header_bytes_received(
+        stream_info.getUpstreamBytesMeter()->headerBytesReceived());
+  }
```

Comment:
could we merge the logic of request properties setting and response properties setting to avoid unnecessary nullptr check?